File: Algorithm/baekjoon/java/src/bj_21557.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;


public class bj_21557 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st ;

        // N
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] arr;
        arr = new int[N];
        // arr
        for (int i=0; i<N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }
        int left = arr[0];
        int right = arr[N-1];

        for (int i=0; i<N-3; i++) {
            if (left>right) {
                left -= 1;
            } else {
                right -= 1;
            }
        }
        int ans = Math.max(left,right) -1;
        System.out.println(ans);
    }
}

File: Algorithm/baekjoon/bj_21557.py
N = int(input())
arr = [*map(int,input().split())]
# 양 끝 폭죽 더미를 제외한 폭죽 더미 하나를 고른다 -> 마지막에는 무조건 양쪽 끝 더미만 남게된다.
left = arr[0]
right = arr[-1]
for _ in range(N-3):
    if left > right : left -= 1
    else : right -= 1
left-=1
right-=1

print(max(left,right))